pilot/pkg/networking/core/v1alpha3/cluster.go

```diff
@@ -1178,17 +1178,13 @@ func applyUpstreamTLSSettings(opts *buildClusterOpts, tls *networking.TLSSetting
 		if cluster.Http2ProtocolOptions != nil {
 			// This is HTTP/2 in-mesh cluster, advertise it with ALPN.
 			if tls.Mode == networking.TLSSettings_ISTIO_MUTUAL {
-				tlsContext.CommonTlsContext.AlpnProtocols = util.ALPNInMeshH2
+				tlsContext.CommonTlsContext.AlpnProtocols = util.ALPNMtlsH2
 			} else {
-				tlsContext.CommonTlsContext.AlpnProtocols = util.ALPNH2Only
+				tlsContext.CommonTlsContext.AlpnProtocols = util.ALPNPlaintextH2Only
 			}
 		} else if tls.Mode == networking.TLSSettings_ISTIO_MUTUAL {
 			// This is in-mesh cluster, advertise it with ALPN.
-			if util.IsTCPMetadataExchangeEnabled(node) {
```

Comment:
Could this cause proxy < 1.5 incompatible?

Reply:
Yes but are we guaranteeing that proxy 1.4 or proxy 1.3/proxy 1.2 can talk to proxy 1.6 ? Do we have tests that block a release if this is not possible?

Reply:
I am not sure if there is any test case . But I remember officially we support 3 releases.

Reply:
proxy to proxy compatiblity and proxy to pilot compatibility are two completely different things. The former we have never broken compatibility since at least 1.0, the latter we have only guaranteed 1 release I think so we support upgrade

Reply:
What will happen if the alpn negotiation succeed (i.e., have an agreement on the alpn) but the server doesn't have metadata exchange filter? @gargnupur

Reply:
this can't happen in 1.6 as all servers have metadata filters since 1.5